internal/provider/data_source_certificate_test.go

```diff
@@ -8,6 +8,53 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
+func TestAccDataSourceCertificate_CertificateContent(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		Providers: testProviders,
+
+		Steps: []resource.TestStep{
+			{
+
+				Config: `
+					data "tls_certificate" "test" {
+					  content = file("testdata/tls_certs/certificate.pem")
+					}
+				`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.#", "1"),
+
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.signature_algorithm", "SHA256-RSA"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.public_key_algorithm", "RSA"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.serial_number", "266244246501122064554217434340898012243"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.is_ca", "false"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.version", "3"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.issuer", "CN=Root CA,O=Test Org,L=Here"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.subject", "CN=Child Cert,O=Child Co.,L=Everywhere"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.not_before", "2019-11-08T09:01:36Z"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.not_after", "2019-11-08T19:01:36Z"),
+					resource.TestCheckResourceAttr("data.tls_certificate.test", "certificates.0.sha1_fingerprint", "61b65624427d75b61169100836904e44364df817"),
+				),
+			},
+			{
+				Config: `
+					data "tls_certificate" "test" {
+					  content = "not a pem"
+					}
+				`,
+				ExpectError: regexp.MustCompile("failed to decode pem content"),
+			},
+			{
+				Config: `
+					data "tls_certificate" "test" {
+					  content = file("testdata/tls_certs/private.pem")
+					}
+				`,
+				ExpectError: regexp.MustCompile("pem must be of type 'CERTIFICATE'"),
+			},
```

Comment:
Could you also add negative cases to prove that it's not possible to use both `url` and `content` at the same time?

Additionally, it would be nicer if the tests were split a bit more, at least to separate "positive" and "negative" cases.